src/NosCore.Controllers/GroupPacketController.cs

```diff
@@ -328,22 +327,27 @@ public void LeaveGroup(PleavePacket __)
 
             if (group.Count > 2)
             {
-                Session.Character.LeaveGroup();
-
                 if (group.IsGroupLeader(Session.Character.CharacterId))
                 {
+                    Session.Character.LeaveGroup();
```

Comment:
no

Reply:
If the player leaves why we check later if he/she is the groupleader? It will never detect him as the groupleader and never pass the if

Reply:
Because we check before not after line 333

Reply:
Mmm... I don't get it. How we checked before if the code was "LeaveGroup(Player A)" and then "Check if Player A is the leader of the group"?